image_finder.py

```diff
@@ -81,65 +76,127 @@ def find_circle_in_image(image_frame: video_stream.TimedFrame, app_context: app_
+            image_frame = video_stream.TimedFrame(median, 200)
+
 
         #edges = cv2.Canny(median, 100, 200)
         #cv2.imshow("mask", edges)
-        #cv2.waitKey()
+        #cv2.waitKey() 
+            
+        ellipses = []
 
         contours, hierarchy = cv2.findContours(median, cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE)
         if(len(contours) ==0):
-            return None, None
-        c = max(contours, key = cv2.contourArea)
-
-        M = cv2.moments(c)
-        cx = int(M['m10']/M['m00'])
-        cy = int(M['m01']/M['m00'])
-
-        bigcircle=(cx, cy, .01)
-        """
-        circles = cv2.HoughCircles(edges, cv2.HOUGH_GRADIENT,
-                                    dp=app_context.circle_context.dp, minDist=100,
-                                    param1=app_context.circle_context.param1,
-                                    param2=app_context.circle_context.param2,
-                                    minRadius=10,maxRadius=200)
-        logger.debug("hough time %d", time.time()-start_time)
-        start_time = time.time()
-
-        if circles is not None:
-            circles = circle_score.filter_by_composite_score(edges, numpy.around(circles[0]), app_context.circle_context.circle_filter_b, app_context.circle_context.circle_filter_m)
-            logger.debug("circle filter time %d", time.time()-start_time)
-            start_time = time.time()
-            circles = numpy.uint16(numpy.around(circles))
-            counter=0
-            big=0
-            for circle in circles[0, :]:
-                counter+=1
+            return None, image_frame
+        for c in contours: 
+            #c = max(contours, key = cv2.contourArea)
+            area = cv2.contourArea(c)
+            if (area > 250):
+                ellipse = cv2.fitEllipse(c)
+                ellipses.append((c,ellipse))
+                if(app_context.color_context.image_show == app_contexts.ImageShow.BLOB_AND_ELLIPSE):
+                    cv2.ellipse(image_frame.frame, ellipse, (36,255,12), 2)
+        
+        if(len(ellipses) ==0):
+            return None, image_frame
 
-                if circle[2] >= big:
-                    big = circle[2]
-                    bigcircle = circle
-        """
-        return bigcircle, image_frame
+        keypoints = blobFilter(median,app_context)
+        x=0
+        y=0
+        if len(keypoints) > 0:
+            x,y =  keypoints[0].pt[0], keypoints[0].pt[1]
+        
+       
+        if( app_context.color_context.image_show == app_contexts.ImageShow.BLOB_AND_ELLIPSE):
+            blank = numpy.zeros((1, 1))  
+            blob = cv2.drawKeypoints(image_frame.frame, keypoints, blank, (0, 0, 255), 
+                            cv2.DRAW_MATCHES_FLAGS_DRAW_RICH_KEYPOINTS) 
+            image_frame = video_stream.TimedFrame(blob, 200)
+
+
+        candidates = []
+        if(x > 0 and y > 0):
+            for contour, e in ellipses:
+                (xc,yc),(d1,d2),angle = e
```

Comment:
formatting, putting spaces in between variables when unpacking a tuple. Also, you can do this in the `for` if you want.